components/eamxx/src/share/property_checks/mass_and_energy_column_conservation_check.cpp

```diff
@@ -278,6 +283,152 @@ PropertyCheck::ResultAndMsg MassAndEnergyColumnConservationCheck::check() const
 }
 
 
+extern "C"
+void eamxx_repro_sum(const Real* send, Real* recv,
+                       Int nlocal, Int nfld, Int fcomm);
+
+
+void MassAndEnergyColumnConservationCheck::global_fixer(const bool & print_debug_info)
+{
+  const auto ncols = m_num_cols;
+  const auto nlevs = m_num_levs;
+
+  //keep dt for fixers with fluxes.
+  //we do not plan to use fluxes in dycore fixer, but the code is already there
+  EKAT_REQUIRE_MSG(!std::isnan(m_dt), "Error! Timestep dt must be set in MassAndEnergyConservationCheck "
+                                      "before running check().");
+  auto dt = m_dt;
+
+  const auto pseudo_density = m_fields.at("pseudo_density").get_view<const Real**> ();
+  const auto T_mid          = m_fields.at("T_mid"         ).get_view<      Real**> ();
+  const auto horiz_winds    = m_fields.at("horiz_winds"   ).get_view<const Real***>();
+  const auto qv             = m_fields.at("qv"            ).get_view<const Real**> ();
+  const auto qc             = m_fields.at("qc"            ).get_view<const Real**> ();
+  const auto qi             = m_fields.at("qi"            ).get_view<const Real**> ();
+  const auto qr             = m_fields.at("qr"            ).get_view<const Real**> ();
+  const auto ps             = m_fields.at("ps"            ).get_view<const Real*>  ();
+  const auto phis           = m_fields.at("phis"          ).get_view<const Real*>  ();
+
+  const auto vapor_flux = m_fields.at("vapor_flux").get_view<const Real*>();
+  const auto water_flux = m_fields.at("water_flux").get_view<const Real*>();
+  const auto ice_flux   = m_fields.at("ice_flux"  ).get_view<const Real*>();
+  const auto heat_flux  = m_fields.at("heat_flux" ).get_view<const Real*>();
+
+  const auto policy = ExeSpaceUtils::get_default_team_policy(ncols, nlevs);
+
+  using namespace ekat::units;
+  using namespace ShortFieldTagsNames;
+
+  //better to get layouts from m_grid, even scalar0d
+  FieldLayout scalar0d({},{});
+  FieldIdentifier s0_fid("s0", scalar0d, kg / kg, m_grid->name());
+  Field res(s0_fid);
+  res.allocate_view();
+  //this does not exist, why? cause it can be a LEV or a COL array?
+  //FieldLayout scalar1d = m_grid->get_1d_scalar_layout();
+  FieldLayout scalar1d{{COL}, {m_num_cols}};
+  FieldIdentifier s1_fid("s1", scalar1d, kg / kg, m_grid->name());
+
+  Field field_version_s1(s1_fid);
+  field_version_s1.allocate_view();
```

Comment:
I think we want to avoid allocations at each time step. There should be a memory buffer for scratch available here, but perhaps I'm wrong.

Reply:
done thanks